### PR TITLE
Couple of minor changes

### PR DESCRIPTION
**Pull request by hkaiser**
- adding stack size detection on Windows
- specialize cache_line_data for items that don't require padding
- fixing HPX_WITH_SWAP_CONTEXT_EMULATION on Windows

### CODE REVIEW

File: libs/concurrency/include/hpx/concurrency/cache_line_data.hpp [hunk 1]
```diff
@@ -40,6 +41,13 @@ namespace hpx {
                            (data_size % threads::get_cache_line_size())) %
                     threads::get_cache_line_size();
             }
+
+            template <typename Data>
+            struct needs_padding
+              : std::integral_constant<bool,
+                    detail::get_cache_line_padding_size(sizeof(Data)) != 0>
+            {
+            };
```
Comment (msimberg) [inline libs/concurrency/include/hpx/concurrency/cache_line_data.hpp]: Just curious, where do we use this with 0-sized data?

File: libs/coroutines/include/hpx/coroutines/detail/get_stack_pointer.hpp [hunk 2]
```diff
@@ -45,6 +48,8 @@ namespace hpx { namespace threads { namespace coroutines { namespace detail {
     }
 }}}}    // namespace hpx::threads::coroutines::detail
 
+#else
+#define HPX_HAVE_THREADS_GET_STACK_POINTER
```
Comment (msimberg) [inline libs/coroutines/include/hpx/coroutines/detail/get_stack_pointer.hpp]: Would you mind inverting the `#if defined(HPX_WINDOWS)` and moving this `#define` up to the top? I think it makes the logic easier to follow.

**Comment by msimberg** [comment]
@hkaiser are #4254 and this the last changes you wanted to have in for 1.4.0?

File: libs/concurrency/include/hpx/concurrency/cache_line_data.hpp (hunk 1 above)
Comment (hkaiser) [reply libs/concurrency/include/hpx/concurrency/cache_line_data.hpp]: It's any data that has a size evenly divisible by the cache line size. I ran into this while working on the new scheduler. MSVC emits a warning for zero-sized arrays.

**Comment by hkaiser** [comment]
> are #4254 and this the last changes you wanted to have in for 1.4.0?

Yes, indeed. Thanks!

File: libs/concurrency/include/hpx/concurrency/cache_line_data.hpp (hunk 1 above)
Comment (msimberg) [reply libs/concurrency/include/hpx/concurrency/cache_line_data.hpp]: Ah right, makes sense. Thanks!

File: libs/coroutines/include/hpx/coroutines/detail/get_stack_pointer.hpp (hunk 2 above)
Comment (hkaiser) [reply libs/coroutines/include/hpx/coroutines/detail/get_stack_pointer.hpp]: Done.

**Pull request merged**